app/v1/support/DesResponseMappingSupport.scala

```diff
@@ -33,6 +35,28 @@ trait DesResponseMappingSupport {
     }
   }
 
+  final def validateCodingOutResponse[T](desResponseWrapper: ResponseWrapper[T], taxYear: String)
+                                        (implicit currentDate: CurrentDate): Either[ErrorWrapper, ResponseWrapper[T]] = {
+    desResponseWrapper.responseData match {
+      case retrieveCodingOutDetailsResponse: RetrieveCodingOutResponse if TaxYearNotEndedValidation.validate(taxYear).isEmpty
+        && checkIfIdDoesntExists(retrieveCodingOutDetailsResponse) =>
+        Left(ErrorWrapper(desResponseWrapper.correlationId, DownstreamError))
+      case _ => Right(desResponseWrapper)
+    }
+  }
+
+  def checkIfIdDoesntExists(response: RetrieveCodingOutResponse): Boolean = {
```

Comment:
if this returns 'true' in the case that the ID does exist, then it should probably be named something like "idsExist"

Reply:
Ignore this